tests/manager/metadata_layer/test_contributor.py

```diff
@@ -31,131 +32,127 @@ def test_from_contribution(self, db: DatabaseTransactionFixture):
         contributor.family_name = "TestAuttie"
         contributor.wikipedia_name = "TestWikiAuth"
         contributor.biography = "He was born on Main Street."
+        contributor.extra[Contributor.BIRTH_DATE] = "2001-01-01"
 
         contributor_data = ContributorData.from_contribution(contribution)
 
         # make sure contributor fields are still what I expect
         assert contributor_data.lc == contributor.lc
         assert contributor_data.viaf == contributor.viaf
-        assert contributor_data.aliases == contributor.aliases
+        assert contributor_data.aliases == tuple(contributor.aliases)
         assert contributor_data.display_name == contributor.display_name
         assert contributor_data.family_name == contributor.family_name
         assert contributor_data.wikipedia_name == contributor.wikipedia_name
         assert contributor_data.biography == contributor.biography
+        assert contributor_data.extra == contributor.extra
 
     def test_lookup(self, db: DatabaseTransactionFixture):
         # Test the method that uses the database to gather as much
         # self-consistent information as possible about a person.
-        def m(*args, **kwargs):
-            return ContributorData.lookup(db.session, *args, **kwargs)
+        lookup = partial(ContributorData.lookup, db.session)
 
         # We know very little about this person.
-        l1, ignore = db.contributor(
+        db.contributor(
             display_name="Ann Leckie",
             sort_name="Leckie, Ann",
         )
+        al = ContributorData(
+            display_name="Ann Leckie", sort_name="Leckie, Ann", roles=[]
+        )
 
         # We know a lot about this person.
-        pkd, ignore = db.contributor(
+        db.contributor(
             sort_name="Dick, Phillip K.",
             display_name="Phillip K. Dick",
             viaf="27063583",
             lc="n79018147",
         )
-
-        def _match(expect, actual):
-            # Verify that two ContributorData objects have the
-            # same db.
-            #
-            # If a value is None in one ContributorData, it must be None
-            # in the other.
-            assert isinstance(actual, ContributorData)
-            assert expect.sort_name == actual.sort_name
-            assert expect.display_name == actual.display_name
-            assert expect.lc == actual.lc
-            assert expect.viaf == actual.viaf
+        pkd = ContributorData(
+            sort_name="Dick, Phillip K.",
+            display_name="Phillip K. Dick",
+            viaf="27063583",
+            lc="n79018147",
+            roles=[],
+        )
```

Comment:
Seems like these two could get out of sync. Not a showstopper, but would be nice to avoid that.